util/whitelist/debugger_whitelist.go

```diff
@@ -0,0 +1,156 @@
+package whitelist
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	whiteListURL   string
+	whiteListCache = cache.New(5*time.Minute, 5*time.Minute)
+	proxyCount     int
+)
+
+func init() {
+	whiteListURL = os.Getenv("WHITELIST_BACKEND_URL")
+	envProxyCount, err := strconv.Atoi(os.Getenv("PROXY_COUNT"))
+	if err != nil || envProxyCount < 0 {
+		envProxyCount = 1000000 // fetch first IP in X-Forwarded-For
```

Comment:
Always fetching the first one will lead to IP forgery, as the first one is user controllable, E.g. `X-Forwarded-For: illegalIp, client, proxy1..proxyN`.
maybe we may set the default value as 0, and this will fetch from r.RemoteAddr default.